backend/src/utils/resourceUtils.ts

```diff
@@ -807,8 +807,10 @@ export const migrateTemplateDisablement = async (
             undefined,
             options,
           )
-          .then((response) => {
-            return response.body as DashboardConfig;
+          .then(() => {
+            return _.merge({}, dashboardConfig, {
+              templateDisablement: templatesDisabled,
+            });
```

Comment:
`templateDisablement` should be assigned as a property of `spec` and not on the root of the config object.

Reply:
`merge` results in a new type that still conforms to `DashboardConfig`. I suggest first cloning the object and then doing an assignment. 
If you provide some type arguments to `merge`, you'll see the type error that is otherwise hidden. eg:
`_.merge<Partial<DashboardConfig>, DashboardConfig, Partial<DashboardConfig>>(...)`

Reply:
@christianvogt You are absolutely right, but I needed to retest it, not sure how but lodash was successfully merging the object here, not sure how that was happening o.O